graphql_api/types/test_analytics/test_analytics.py

```diff
@@ -1,24 +1,318 @@
+            edges=[],
+            total_count=0,
+            page_info={},
+        )
+
+    if term:
+        table = table.filter(pl.col("name").str.starts_with(term))
+
+    if testsuites:
+        table = table.filter(
+            pl.col("testsuite").is_not_null() & pl.col("testsuite").is_in(testsuites)
+        )
+
+    if flags:
+        table = table.filter(
+            pl.col("flags").list.eval(pl.element().is_in(flags)).list.any()
+        )
+
+    match parameter:
+        case TestResultsFilterParameter.FAILED_TESTS:
+            table = table.filter(pl.col("total_fail_count") > 0)
+        case TestResultsFilterParameter.FLAKY_TESTS:
+            table = table.filter(pl.col("total_flaky_fail_count") > 0)
+        case TestResultsFilterParameter.SKIPPED_TESTS:
+            table = table.filter(
+                pl.col("total_skip_count") > 0 & pl.col("total_pass_count") == 0
+            )
+        case TestResultsFilterParameter.SLOWEST_TESTS:
+            table = table.filter(
+                pl.col("avg_duration") >= pl.col("avg_duration").quantile(0.95)
+            ).top_k(100, by=pl.col("avg_duration"))
+
+    total_count = table.height
+
+    def ordering_expression(cursor_value: CursorValue, is_forward: bool) -> pl.Expr:
+        if is_forward:
+            ordering_expression = (
+                pl.col(ordering.value) > cursor_value.ordered_value
+            ) | (
+                (pl.col(ordering.value) == cursor_value.ordered_value)
+                & (pl.col("name") > cursor_value.name)
+            )
+        else:
+            ordering_expression = (
+                pl.col(ordering.value) < cursor_value.ordered_value
+            ) | (
+                (pl.col(ordering.value) == cursor_value.ordered_value)
+                & (pl.col("name") > cursor_value.name)
+            )
+        return ordering_expression
+
+    if after:
+        if ordering_direction == OrderingDirection.ASC:
+            is_forward = True
+        else:
+            is_forward = False
+
+        cursor_value = decode_cursor(after, ordering)
+        if cursor_value:
+            table = table.filter(ordering_expression(cursor_value, is_forward))
+    elif before:
+        if ordering_direction == OrderingDirection.DESC:
+            is_forward = True
+        else:
+            is_forward = False
+
+        cursor_value = decode_cursor(before, ordering)
+        if cursor_value:
+            table = table.filter(ordering_expression(cursor_value, is_forward))
+
+    table = table.sort(
+        [ordering.value, "name"],
+        descending=[ordering_direction == OrderingDirection.DESC, False],
+    )
+
+    if first:
+        page_elements = table.slice(0, first)
+    elif last:
+        page_elements = table.reverse().slice(0, last)
+    else:
+        page_elements = table
+
+    rows = [TestResultsRow(**row) for row in page_elements.rows(named=True)]
+
+    page: list[dict[str, str | TestResultsRow]] = [
```

Comment:
this should be pages right? More than one

Reply:
no, this is a single page, since it's after all the ordering and filtering